guides/javascript.md

```diff
@@ -0,0 +1,264 @@
+
+## Use a verb phrase for function/method names
+
+When naming a function or method, describe the action, not the outcome.
+
+🚫
+
+``` javascript
+function formattedChangelog() {
+  // ...
+}
+```
+
+✅
+
+``` javascript
+function formatChangelog() {
+  // ...
+}
+```
+
+For a function or method that returns a boolean, prefix the name with a descriptive verb so that readers don't confuse it with a boolean variable.
+
+🚫
+
+``` javascript
+class NetworkController {
+  isEIP1559Compatible() {
+    // ...
+  }
+}
+```
+
+✅
+
+``` javascript
+class NetworkController {
+  determineEIP1559Compatibility() {
+    // ...
+  }
+}
+```
+
+### Read more
+
+- [Google's JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html#naming-method-names)
+- [.NET Fundamentals](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods)
+- [Kotlin Coding Conventions](https://kotlinlang.org/docs/coding-conventions.html#choose-good-names)
+- [Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage)
+
+## Use a verb phrase for boolean variable names that do not involve secondary objects
+
+Sometimes a variable that is intended to hold a boolean value does not have an explicit subject, but represents the context where the variable is defined (e.g., a class or an entire file). When naming such a variable, use a statement that describes the state of the subject, minus the name of the subject itself. Usually this means prefixing the name with a form of "to be" or "to have" (e.g. `is*`, `has*`), but you may find it more readable to use past or future tense and/or a modal verb such as `should`.
+
+🚫
+
+``` javascript
+const removed = false;
+```
+
+✅
+
+``` javascript
+// Any of these would do
+const isRemoved = false;
+const wasRemoved = false;
+const hasBeenRemoved = false;
+const shouldBeRemoved = false;
+```
+
+If the name represents a negative statement, reword it into a positive statement and inverting the value assigned to the value.
+
+🚫
+
+``` javascript
+const notEnoughGas = false;
+```
+
+✅
+
+``` javascript
+const hasEnoughGas = true;
+```
+
+Take special note of variables which are created via React's `useState` hook.
+
+🚫
+
+``` javascript
+const [removed, setRemoved] = useState(false);
+```
+
+✅
+
+``` javascript
+const [isRemoved, setIsRemoved] = useState(false);
+```
+
+## 💡 Place names of secondary concepts first in boolean variable names
+
+Sometimes a boolean variable describes a subject that's different than the context where the variable is defined:
+
+``` javascript
+const ownedAccount = Boolean(ownedAccountName);
+```
+
+In naming such a variable, there are a couple of different routes you could take.
+
+One way is to always place the verb at the beginning:
```

Comment:
This example was a little hard to understand for me. I'm not sure what the subject or the context are supposed to be here.

However, the later examples seemed very clear and easy to understand. Maybe this would be simpler if we got to these examples sooner? e.g. we could skip straight to the `isRecipientOwnedAccount` example, skipping these first two.

Something like:

```suggestion
When naming a boolean variable that includes a subject, the previous guideline suggests that you can place the verb at the beginning:
```

Reply:
Hah, just noticed that we're calling "is" a verb here. That doesn't seem right.

Reply:
I also wasn't quite sure, but I remember looking this up before typing it. But I've looked it up again to be double-sure.

It looks like "is" is a verb, but it's special. It doesn't do anything but tie the subject and object together, so sometimes it's called a "linking" or "copula" verb: https://en.wikipedia.org/wiki/Linking_verb

Reply:
Oops I didn't mean to resolve this but I hope that makes sense ^

Reply:
Yep, makes sense!